wavefront/server/plugins/authenticator/authenticator/microsoft_adfs/authenticator.py

```
@@ -0,0 +1,324 @@
+        # The id_token arrived over TLS from an endpoint we just successfully
+        # authenticated against, so we trust the channel and skip signature
+        # verification (which would require fetching JWKS per IdP).
+        try:
+            segments = id_token.split('.')
+            if len(segments) < 2:
+                return None
+            payload_b64 = segments[1]
+            padding = '=' * (-len(payload_b64) % 4)
+            payload_bytes = base64.urlsafe_b64decode(payload_b64 + padding)
+            return json.loads(payload_bytes)
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)